### DISCUSSION TITLE
how to make a container stick to window right

### DISCUSSION THREAD

**Discussion by dashessmith** [fyne-io/fyne#3221] [Q&A] [answered]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

![image](https://user-images.githubusercontent.com/75690316/186291069-c460a2f9-ddf7-4412-b6ab-7742f29e2aad.png)

in the begging it's ok , but after change blue part (a vbox), the green part becomes weird

this code seperate the whole into left & right and stick right to most right
`func (z *MiddleRenderer) Layout(size fyne.Size) {
	// /* 剩余空间全给市场部分 */
	// z.marketPanel.Resize(fyne.Size{
	// 	Width:  size.Width - z.ctrlPanel.Size().Width,
	// 	Height: z.marketPanel.Size().Height,
	// })
	// //z.marketPanel.Move(fyne.Position{X: 0, Y: 0})

	// /* 仓位部分靠近底部 */
	// z.posiPanel.Resize(fyne.Size{
	// 	Width:  size.Width - z.ctrlPanel.Size().Width,
	// 	Height: z.posiPanel.Size().Height,
	// })
	// //z.posiPanel.Move(fyne.Position{X: 0, Y: size.Height - z.posiPanel.Size().Height})

	/* 行情 + 仓位 */
	/* VSPLIT PART */
	tickPanelSize := fyne.Size{
		Width:  size.Width - z.ctrlPanel.Size().Width,
		Height: size.Height,
	}
	z.tickPanel.Resize(tickPanelSize)
	z.tickPanel.minSize = tickPanelSize

	/* 控制区域 */
	/* STICK TO RIGHT */
	z.ctrlPanel.Move(fyne.Position{X: gapp.fyneWindow.Content().Size().Width - z.ctrlPanel.Size().Width, Y: 0})

	// z.container.Refresh()
}`

but after change the positions vbox, things get weird
`{
		// middleSize := z.middlePanel.Size()
		z.positionsList.RemoveAll()
		for _, bpos := range w.Positions {
			for _, pos := range bpos {
				if pos == nil || pos.Qty <= 0 {
					continue
				}
				log.Printf("ui new pos entry")
				entry := newPostionEntry()
				entry.Coin = pos.TradePair
				entry.OpenPrice = pos.OpenPrice
				entry.Qty = pos.Qty
				entry.Profit = .0 // todo
				entry.Direct = pos.Direct
				z.positionsList.Add(entry)
			}
		}
		// z.middlePanel.Move(fyne.Position{X:0, Y:0})
		// z.middlePanel.UpdateSize(middleSize)
		// z.middlePanel.Show()

	}`

### How to reproduce

see description

### Example code

see description

### Fyne version

see description

### Go compiler version

see description

### Operating system

Windows

### Operating system version

see description

### Additional Information

see description

**Comment by dashessmith** [comment]
![image](https://user-images.githubusercontent.com/75690316/186293520-fbd091a2-6553-4031-a6a9-ecfb9e2e30d8.png)

**Comment by dashessmith** [comment]
inside left part
![image](https://user-images.githubusercontent.com/75690316/186293773-f5b20768-bf02-4a4a-8b0d-65918b166799.png)

**Comment by Jacalz** [comment]
Please use GitHub discussions, or one of the other support methods mentioned on https://fyne.io/support for questions. We want to use the issue tracker for bug reports and feature requests only. I will transfer this over to a discussion.

**Comment by andydotxyz** [comment, marked as answer]
You should remove all of those `Max` containers, they do nothing (a Max with 1 item is the same as not using a container at all, widgets will always fill the space available.

Instead of all your manual code I would use `container.NewBorder(nil, nil, nil, right, content)`, so that the right component takes up space on the right and content fills the remaining space.
https://developer.fyne.io/explore/layouts
